src/main/java/org/geekhub/json/JsonSerializer.java

```diff
@@ -70,7 +74,38 @@ public static Object serialize(Object o) {
      * @throws InstantiationException
      */
     private static JSONObject toJsonObject(Object o) throws Exception {
-        //implement me
-        return null;
+        JSONObject jsonObject = new JSONObject();
+
+        Class argClass = o.getClass();
+        Field fields[] = argClass.getDeclaredFields();
+        for (Field field : fields) {
+            field.setAccessible(true);
+            if (field.isAnnotationPresent(Ignore.class) || field.isSynthetic()) {
+                continue;
+            }
+
+            String fieldName = field.getName();
+            Class<?> fieldType = field.getType();
+            Object fieldValue = field.get(o);
+
+            UseDataAdapter dataAdapter = field.getAnnotation(UseDataAdapter.class);
+            if (dataAdapter != null) {
+                Class adapterClass = (Class) dataAdapter.value();
+                jsonObject.put(fieldName, ((JsonDataAdapter) adapterClass.newInstance()).toJson(fieldValue));
+            }
+
+            if (Collection.class.isAssignableFrom(fieldType)) {
```

Comment:
Could you show to me use-case, because now I think, that my code will do it in right way.